test.md

```diff
@@ -0,0 +1 @@
+This file should be ignored by git.
```

Comment:
- [ ] Yes this file should be ignored by git, and I like that you put it on .gitignore, that's commendable, however it was not ignored by git because by the time you created it, you committed it before you add it to gitignore, so because of that git will not care about what you add on gitignore it cares so much about your commits, so it cannot remove it from your working tree unless you tell it that you committed this file by mistake, but now you decided to remove it from your working tree.

:grey_question:  Then how to tell it to remove it now? :question: :facepalm: 

:+1: Just run this command: `git rm -f test.md` and then commit the changes, after that command, you will see that now the file `test.md` is no longer present even on your local directory, since now it's on the gitignore, then you can now freely create it, it can't go back in your commits even though you commit thousands time.